### PR TITLE
Feature/#1890 Fix a bug when an player is not visible in chat list after removing from friend list

### LINKED ISSUE #1890: Adding then removing player from friend list removes them from chat list
Removing a player from the chat list removes their name from the chat list entirely

**Reproduce**
Add then remove player from friend list

**Expected behavior**
They should move back to the others list

**OS**
Windows

### PR DESCRIPTION
**Pull request by Marc-Spector**
Fixes #1890

### CODE REVIEW

**Comment by Marc-Spector** [comment]
I have additionally added two very important tests (`testWhenUserAddPlayerToFriendlyListByAccidentallyAndCancelsActionImmediately` and `testWhenUserAddPlayerToFoeListByAccidentallyAndCancelsActionImmediately`). Without my fix, these tests will fail.

**Comment by Marc-Spector** [comment]
The PR is ready to review

File: src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java [hunk 1]
```diff
@@ -13,6 +13,15 @@
...
   ChatUserCategory category;
   ChatChannelUser user;
+
+  public boolean isUserItem() {
+    return user != null;
+  }
+
+  public boolean isCategoryItem() {
+    return category != null;
+  }
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java]: These are only ever used followed by an .equals().
Wouldn't it be easier to write a
`public boolean contains(Object item)` ?

File: src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java [hunk 2]
```diff
@@ -13,6 +13,15 @@
  */
 @Value
 class CategoryOrChatUserListItem {
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java:15]: While refactoring this class, why not offer either individual constructors only taking a category or a user or offering static builder function `of(ChatUserCategory category)` and  `of(ChatChannelUser user)`

File: src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java (hunk 1 above)
Comment (Marc-Spector) [reply src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java]: Ok, it is a good idea.

Comment (Marc-Spector) [reply src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java]: I will not write this method. There is an easier way.

Comment (Marc-Spector) [reply src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java]: See please https://github.com/FAForever/downlords-faf-client/pull/2022/files#diff-55c0277ffe170fd8988215bc63e3d03909588a2d72fb0fa2e9cca785cdd12bb9R500

Comment (Brutus5000) [reply src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java]: Why did you drop `@Value`? You could just configure the default constructor to be private and call the private constructor from your custom . If you drop @Value you loose ToString, Equals and HashCode features from Lombok

Comment (Marc-Spector) [reply src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java]: Reverted.

**Review by Sheikah45 — APPROVED** [review]
Looks good to me

**Pull request merged**